build error `*mut cuda_driver_sys::CUctx_st` cannot be sent between threads safely

It is very strange that when I use sea-orm's mysql feature and rust-gpu-tool to compile together, an error will pop up， but when i remove mysql feature, build can be success.

this is a mininal repro example.
create empty project and modify cargo toml  like below
```toml
[package]
name = "testp"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
rust-gpu-tools = { version = "0.6", default-features = false }

[dependencies.sea-orm]
version ="0.6.0"
features = [
    "runtime-tokio-native-tls",
    "sqlx-mysql",
]
default-features = false

[features]
default = ["rust-gpu-tools/cuda"]
```
cargo build, got error like below
```
root@200-6:~/lijunlong/testp# cargo build
   Compiling temp-env v0.2.0
   Compiling chrono v0.4.19
   Compiling uuid v0.8.2
   Compiling futures-executor v0.3.21
   Compiling sqlx-core v0.5.13
   Compiling rust-gpu-tools v0.6.0
   Compiling futures v0.3.21
error[E0277]: `*mut cuda_driver_sys::CUctx_st` cannot be sent between threads safely
  --> /root/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-7a04d2510079875b/rust-gpu-tools-0.6.0/src/device.rs:29:1
   |
29 | / lazy_static! {
30 | |     // The owned CUDA contexts are stored globally. Each devives contains an unowned reference,
31 | |     // so that devices can be cloned.
32 | |     static ref DEVICES: (Vec<Device>, cuda::utils::CudaContexts) = build_device_list();
33 | | }
   | |_^ `*mut cuda_driver_sys::CUctx_st` cannot be sent between threads safely
   |
   = help: within `fil_rustacuda::context::Context`, the trait `Send` is not implemented for `*mut cuda_driver_sys::CUctx_st`
   = note: required because it appears within the type `fil_rustacuda::context::Context`
   = note: required because of the requirements on the impl of `Send` for `Unique<fil_rustacuda::context::Context>`
   = note: required because it appears within the type `alloc::raw_vec::RawVec<fil_rustacuda::context::Context>`
   = note: required because it appears within the type `Vec<fil_rustacuda::context::Context>`
note: required because it appears within the type `CudaContexts`
  --> /root/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-7a04d2510079875b/rust-gpu-tools-0.6.0/src/cuda/utils.rs:14:19
   |
14 | pub(crate) struct CudaContexts(Vec<rustacuda::context::Context>);
   |                   ^^^^^^^^^^^^
   = note: required because it appears within the type `(Vec<device::Device>, CudaContexts)`
   = note: required because of the requirements on the impl of `Sync` for `spin::once::Once<(Vec<device::Device>, CudaContexts)>`
   = note: required because it appears within the type `lazy_static::lazy::Lazy<(Vec<device::Device>, CudaContexts)>`
   = note: shared static variables must have a type that implements `Sync`
   = note: this error originates in the macro `__lazy_static_create` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `rust-gpu-tools` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

but if remove sqlx-mysql featurem build can be pass, 
how can i deal with this problem

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,7 +19,6 @@ cuda = ["rustacuda"]
 dirs = "5.0.0"
 sha2 = "0.10"
 thiserror = "1.0.10"
-lazy_static = "1.2"
 log = "0.4.11"
 hex = "0.4.3"
 

--- a/src/cuda/utils.rs
+++ b/src/cuda/utils.rs
@@ -7,12 +7,13 @@ use crate::device::{PciId, Vendor};
 use crate::error::{GPUError, GPUResult};
 
 // NOTE vmx 2021-04-14: This is a hack to make sure contexts stay around. We wrap them, so that
-// `Sync` can be implemented. `Sync` is needed for lazy static. These contexts are never used
-// directly, they are only accessed through [`cuda::Device`] which contains an `UnownedContext`.
-// A device cannot have an own context itself, as then it couldn't be cloned, but that is needed
-// for creating the kernels.
+// `Sync` and `Send` can be implemented. `Sync` and `Send` is needed for once_cell. These contexts
+// are never used directly, they are only accessed through [`cuda::Device`] which contains an
+// `UnownedContext`. A device cannot have an own context itself, as then it couldn't be cloned,
+// but that is needed for creating the kernels.
 pub(crate) struct CudaContexts(Vec<rustacuda::context::Context>);
 unsafe impl Sync for CudaContexts {}
+unsafe impl Send for CudaContexts {}
 
 /// The PCI-ID is the combination of the PCI Bus ID and PCI Device ID.
 ///

--- a/src/device.rs
+++ b/src/device.rs
@@ -1,9 +1,9 @@
 use std::fmt;
 
-use lazy_static::lazy_static;
 use log::debug;
 #[cfg(all(feature = "opencl", feature = "cuda"))]
 use log::warn;
+use once_cell::sync::Lazy;
 
 use std::convert::TryFrom;
 use std::mem;
@@ -25,19 +25,15 @@ const AMD_DEVICE_ON_APPLE_VENDOR_ID: u32 = 0x1021d00;
 const NVIDIA_DEVICE_VENDOR_STRING: &str = "NVIDIA Corporation";
 const NVIDIA_DEVICE_VENDOR_ID: u32 = 0x10de;
 
+// The owned CUDA contexts are stored globally. Each devives contains an unowned reference, so
+// that devices can be cloned.
 #[cfg(feature = "cuda")]
-lazy_static! {
-    // The owned CUDA contexts are stored globally. Each devives contains an unowned reference,
-    // so that devices can be cloned.
-    static ref DEVICES: (Vec<Device>, cuda::utils::CudaContexts) = build_device_list();
-}
+static DEVICES: Lazy<(Vec<Device>, cuda::utils::CudaContexts)> = Lazy::new(build_device_list);
 
+// Keep it as a tuple as the CUDA case, so that the using `DEVICES` is independent of the
+// features set.
 #[cfg(all(feature = "opencl", not(feature = "cuda")))]
-lazy_static! {
-    // Keep it as a tuple as the CUDA case, so that the using `DEVICES` is independent of the
-    // features set.
-    static ref DEVICES: (Vec<Device>, ()) = build_device_list();
-}
+static DEVICES: Lazy<(Vec<Device>, ())> = Lazy::new(build_device_list);
 
 /// The PCI-ID is the combination of the PCI Bus ID and PCI Device ID.
 ///

